src/js/components/Heading/StyledHeading.js

```diff
@@ -24,8 +23,13 @@ const sizeStyle = props => {
         props.theme.global.breakpoints[headingTheme.responsiveBreakpoint];
       if (breakpoint) {
         const responsiveData =
-          headingTheme.level[Math.min(props.level + 1, 4)][size];
-        if(responsiveData) {
+          headingTheme.level[
+            Math.min(
+              props.level + 1,
+              Math.max(...Object.keys(headingTheme.level)),
```

Comment:
I think this will always be 6 since the base theme will always have the level 6 defined, is that we want?

Reply:
Yes, it will always be 6. Going off of @ericsoderberghp suggestion. I assume this will support any future heading levels you might create later.

Reply:
Maybe I'm going a little too far here, but we might want to make sure the users didn't override the theme with empty objects of 5,6, just so they won't be considered.
Should we reach inside the level to see if the 'medium' size (default) or the required size is even there?

Reply:
It would be friendlier to be defensive. Use the next level down, until we get to the last level we have, which could be different for different themes.

Reply:
The way the code currently is
```
Math.min(props.level + 1, Math.max(...Object.keys(headingTheme.level))
```
will return the lowest level value(largest number) that isn't an empty object.

Reply:
Can you please add a test that proves and enforces it?

Reply:
Interesting, you can assign object to undefined instead of undefined.

Reply:
im not sure what you mean, could you elaborate?

Reply:
passing in undefined in the level value will cause an error. Although, I'm not sure what we should do about this. To me, this seems like a problem with user error. I could go to the next lowest level in heading that isn't empty, but if undefined is passed through any key in level(like font, or size), I will also receive an error. 

Everything that causes an error:
```
'6': undefined,
```

```
'6': {
  font: undefined,
  small: {
    size: '140px',
    height: '20px',
    maxWidth: '336px',
  },
  medium: {
    size: '140px',
    height: '20px',
    maxWidth: '336px',
  },
  large: {
    size: '140px'
    height: '20px',
    maxWidth: '336px',
  },
  xlarge: {
    size: '140px',
    height: '20px',
    maxWidth: '336px',
  },
},
```

```
'6': {
  font: {},
  small: {
    size: '140px',
    height: '20px',
    maxWidth: '336px',
  },
  medium: undefined ,
  large: {
    size: '140px'
    height: '20px',
    maxWidth: '336px',
  },
  xlarge: {
    size: '140px',
    height: '20px',
    maxWidth: '336px',
  },
},
```
doesn't error - uses default theme:
```
'6': {},
```

I could be wrong, but I don't think there's a way to enforce the correct theme shape in this instance(there are a lot of ways that i can make this fail). There's a lot that a user can do wrong, and I think this would be better addressed with complete and clear documentation.

Reply:
@halocline can you please advise Ian on how to apply a test or a fix for this issue and move this PR forward.

Reply:
@IanKBovard What do you think about an approach where we probe for the next level down, if it is defined, use it; if undefined, use current level?

Could look something like: 
```
const responsiveData = headingTheme.level[props.level + 1]
          ? headingTheme.level[props.level + 1][size]
          : headingTheme.level[props.level][size];
```

From a test standpoint, I think we could add Jest tests for theme where we pass in each of your scenarios above, perhaps at varying lower levels as well.

Regarding the font definitions, I think we may need to place an additional check for heading.level existing at where the fontFamily is being defined on line 53.